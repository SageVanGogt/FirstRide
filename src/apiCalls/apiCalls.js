import { geoKey } from './../apiKey';

const signinUser = async (user) => {
  const url = 'http://localhost:3000/api/users/';
  const init = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, init);
  const profile = await response.json();
  return profile;
}; 

const signupUser = async (user) => {
  const url = 'http://localhost:3000/api/users/new';
  const body = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const userRecieved = await response.json();
  return userRecieved;
};

const fetchDestination = async (name) => {
  const url = `http://localhost:3000/api/locations/${name}`;
  const response = await fetch(url);
  const destination = await response.json();
  return destination;
};

const fetchRides = async (destination) => {
  const url = `http://localhost:3000/api/rides/get/${destination}`;
  const body = {
    method: 'POST', 
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const rides = await response.json();
  return rides;
};

const fetchPickups = async (location) => {
  const url = `http://localhost:3000/api/pickup/get/${location}`;
  const body = {
    method: "POST", 
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const pickups = await response.json();
  return pickups;
};

const fetchGeocode = async (address) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${address}&key=${geoKey}`;  
  try {
    const response = await fetch(url);
    const locationData = await response.json();
    if (!locationData) {
      return new Error('error')
    }
    return locationData;
  } catch (error) {
    throw error;
  }
}; 

const submitNewRide = async (ride) => {
  const url = `http://localhost:3000/api/rides/new`;
  const body = {
    method: 'POST',
    body: JSON.stringify(ride),
    headers: {
      "Content-Type": "application/json"        
    }
  };
  const response = await fetch(url, body);
  const rideId = await response.json();
  return rideId;
};

const submitNewPickup = async (location) => {
  const url = `http://localhost:3000/api/pickup/new`;
  const body = {
    method: 'POST',
    body: JSON.stringify(location),
    headers: {
      "Content-Type": "application/json"        
    }
  };
  const response = await fetch(url, body);
  const pickupId = await response.json();
  return pickupId;
};

const fetchRidesPassengers = async (locationId) => {
  const url = `http://localhost:3000/api/rides_passengers/get/passengers/${locationId}`;
  const body = {
    method: 'POST', 
    headers: {
      'Content-Type': 'application/json'
    }
  };
  const response = await fetch(url, body);
  const passengers = await response.json();
  return passengers;
};

const postRidesPassengers = async (newPassenger) => {
  const url = `http://localhost:3000/api/rides_passengers/new`;   
  const body = {
    method: "POST",
    body: JSON.stringify(newPassenger),
    headers: {
      'Content-Type': 'application/json'
    }
  };
  const response = await fetch(url, body);
  const ridePassenger = await response.json();
  return ridePassenger;
};

const postNewProfile = async (profile) => {
  const url = `http://localhost:3000/api/profiles/new`;
  const body = {
    method: "POST",
    body: JSON.stringify(profile),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const profileInfo = await response.json();
  return profileInfo;
};

const removePassengerRide = async (rideId, passengerId, destinationId) => {
  const url = `http://localhost:3000/api/rides/${rideId}/passengers/${passengerId}/destination/${destinationId}`; 
  const body = {
    method: "DELETE",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const deletedId = await response.json();
  return deletedId;
};

const fetchUserRides = async (userId) => {
  const url = `http://localhost:3000/api/rides_passengers/get/rides/${userId}`;
  const body = {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const rides = response.json();
  return rides;
};

const fetchRidesFromUser = async (rideId) => {
  const url = `http://localhost:3000/api/rides/${rideId}/get/user/`;
  const body = {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const ridesInfo = await response.json();
  return ridesInfo;
};

const reverseGeoCode = async (lat, lng) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lng}&key=${geoKey}`;
  const response = await fetch(url);
  const geoData = await response.json();
  return geoData;
}

export {
  signinUser,
  signupUser,
  fetchDestination,
  fetchRides,
  fetchPickups,
  fetchGeocode,
  submitNewRide,
  submitNewPickup,
  fetchRidesPassengers,
  postRidesPassengers,
  postNewProfile,
  removePassengerRide,
  fetchUserRides,
  fetchRidesFromUser,
  reverseGeoCode
};