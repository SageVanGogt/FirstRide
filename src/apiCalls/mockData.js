export const mockUser = {
  id:1,
  user_id:"456",
  user_img:"http://us.cdn281.fansshare.com/photos/channingtatum/bfhgoi-ccaagvrc-large-fat-44506770.jpg",
  bio:"I like to eat",
  rating:"Solid 7"
}