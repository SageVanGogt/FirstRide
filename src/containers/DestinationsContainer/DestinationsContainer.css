.destinations {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100vw;
  margin-top: 70px;
  height: calc(100vh - 90px);
}

.redrocks-container {
  display: flex;
  justify-content: center;
  align-items: center;  
  background-image: url('./../../assets/red-rocks.jpeg');
  background-repeat: no-repeat;
  background-size: cover;
  opacity: .4;
  transition: .5s;
  filter: grayscale(100%);  
  width: 100%;
  height: 100%;  
}

.breck-container {
  display: flex;
  justify-content: center;
  align-items: center;
  background-repeat: no-repeat;  
  background-image: url('./../../assets/breck.jpeg');
  background-size: cover; 
  filter: grayscale(100%); 
  transition: .5s;  
  opacity: .5;
  width: 100%;
  height: 100%;
}

.redrocks-container:hover, 
.breck-container:hover {
  opacity: 1;
  filter: grayscale(0%);
}

.location-pin {
  font-size: 3rem;
  color: white;
  text-decoration: none;
  padding: 5px 15px;
  border: 2px solid transparent;
  border-radius: 5px;
  transition: .3s;
  visibility: hidden;
}

.breck-container:hover .location-pin {
  visibility: visible;
  font-size: 3rem;
  text-decoration: none;
  color: white;
  border: 2px solid white; 
}

.redrocks-container:hover .location-pin {
  visibility: visible;
  font-size: 3rem;
  text-decoration: none;
  color: white;
  border: 2px solid white; 
}

.landing-instructions {
  position: absolute;
  z-index: 1;
  top: 200px;
  text-align: center;
  font-size: 4rem;
  text-shadow: 0px 4px 3px rgba(0,0,0,0.4),
             0px 8px 13px rgba(0,0,0,0.1),
             0px 18px 23px rgba(0,0,0,0.1);
  color: white;
  font-family: 'Roboto', sans-serif; 
  /* font-family: 'Work Sans', sans-serif;   */
  font-weight: 800;
  letter-spacing: 1pt;
  pointer-events: none;
}
