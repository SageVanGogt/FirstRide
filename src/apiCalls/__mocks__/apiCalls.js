import * as MOCK from './../mockData';

const signinUser = jest.fn().mockImplementation(() => 
  Promise.resolve({
    user: [{
    "id":1,
    "user_id":"456",
    "user_img":"http://us.cdn281.fansshare.com/photos/channingtatum/bfhgoi-ccaagvrc-large-fat-44506770.jpg",
    "bio":"I like to eat",
    "rating":"Solid 7"
    }]
  })
);

const signupUser = jest.fn().mockImplementation(() => Promise.resolve({
  "id": 1
}));

const fetchDestination = jest.fn().mockImplementation(() => Promise.resolve({
  "id": 1,
  "location_name": "Red Rocks",
  "location_lat_lng": "12341, 123412"
}));

const fetchRides = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockRides));

const fetchPickups = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockPickups));

const fetchGeocode = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockGeoInfo));

const submitNewRide = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

const submitNewPickup = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

const postRidesPassengers = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

export {
  signinUser,
  signupUser,
  fetchDestination,
  fetchRides,
  fetchPickups,
  fetchGeocode,
  submitNewPickup,
  submitNewRide,
  postRidesPassengers
}