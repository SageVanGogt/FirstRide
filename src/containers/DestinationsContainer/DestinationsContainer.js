import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { fetchDestination } from '../../apiCalls/apiCalls';
import { setDestination } from './../../actions/destination';
import './DestinationsContainer.css';

export class DestinationsContainer extends Component {
  constructor(props) {
    super(props);
  }

  handleSelectDestination = async (event) => {
    const { name } = event.target;
    try {
      const response = await fetchDestination(name);
      const finalDestination = response.locations[0];
      this.props.setDestination(finalDestination);
    } catch (err) {
      return err;
    }
  }

  render() {
    return (
      <div className="destinations">
        <h1 className="landing-instructions">
          Choose a destination to see available rides.
        </h1>
        <div className="redrocks-container">
          <NavLink  
            to='/rides'
            onClick={this.handleSelectDestination} 
            name="Red Rocks"
            className="location-pin">
            Red Rocks
          </NavLink>
        </div>
        <div className="breck-container">
          <NavLink
            to='/rides'
            onClick={this.handleSelectDestination} 
            name="Breckenridge"
            className="location-pin">
            Breckenridge
          </NavLink>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setDestination: (data) => dispatch(setDestination(data))
});

DestinationsContainer.propTypes = {
  setDestination: PropTypes.func,
  handleSelectDestination: PropTypes.func
};

export default connect(null, mapDispatchToProps)(DestinationsContainer);