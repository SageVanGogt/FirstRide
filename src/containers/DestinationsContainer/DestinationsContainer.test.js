import React from 'react';
import { shallow } from 'enzyme';
import { DestinationsContainer, mapDispatchToProps } from './DestinationsContainer';
import { fetchDestination } from './../../apiCalls/apiCalls';

jest.mock('./../../apiCalls/apiCalls');

describe('DestinationsContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<DestinationsContainer />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('handleSelectDestination', () => {
    
    it('should call fetchDestination with the correct params', async () => {
      const mockEvent = {
        target: {
          name: 'Red Rocks'
        }
      };
      const expected = 'Red Rocks';
      Promise.resolve(wrapper.instance().handleSelectDestination(mockEvent));
      
      expect(fetchDestination).toHaveBeenCalledWith(expected);
    })
  })

  describe('mapDispatchToProps', () => {
    it('should be called with the correct params', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const expected = {
        type: 'SET_DESTINATION',
        destination: {
          id: 1, 
          location_name: 'Red Rocks', 
          location_lat_lng: '39.6654, 105.2057'
        }
      }
      const mockDestination = {
        id: 1, 
        location_name: 'Red Rocks', 
        location_lat_lng: '39.6654, 105.2057'
      }
      mappedProps.setDestination(mockDestination);

      expect(mockDispatch).toHaveBeenCalledWith(expected);
    })
  })
})