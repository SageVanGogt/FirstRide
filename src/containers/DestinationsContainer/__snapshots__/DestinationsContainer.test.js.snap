// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationsContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DestinationsContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="location-container"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="location-pin"
                    name="Red Rocks"
                    onClick={[Function]}
                    to="/rides"
          >
                    Red Rocks
                    <div
                              className="pulse"
                    />
          </NavLink>
</div>,
        <div
          className="location-container"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="location-pin"
                    name="Breckenridge"
                    onClick={[Function]}
                    to="/rides"
          >
                    Breckenridge
          </NavLink>
          <div
                    className="pulse"
          />
</div>,
      ],
      "className": "destinations",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="location-pin"
            name="Red Rocks"
            onClick={[Function]}
            to="/rides"
>
            Red Rocks
            <div
                        className="pulse"
            />
</NavLink>,
          "className": "location-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": Array [
              "Red Rocks",
              <div
                className="pulse"
/>,
            ],
            "className": "location-pin",
            "name": "Red Rocks",
            "onClick": [Function],
            "to": "/rides",
          },
          "ref": null,
          "rendered": Array [
            "Red Rocks",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "pulse",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="location-pin"
              name="Breckenridge"
              onClick={[Function]}
              to="/rides"
>
              Breckenridge
</NavLink>,
            <div
              className="pulse"
/>,
          ],
          "className": "location-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Breckenridge",
              "className": "location-pin",
              "name": "Breckenridge",
              "onClick": [Function],
              "to": "/rides",
            },
            "ref": null,
            "rendered": "Breckenridge",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "pulse",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="location-container"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="location-pin"
                        name="Red Rocks"
                        onClick={[Function]}
                        to="/rides"
            >
                        Red Rocks
                        <div
                                    className="pulse"
                        />
            </NavLink>
</div>,
          <div
            className="location-container"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="location-pin"
                        name="Breckenridge"
                        onClick={[Function]}
                        to="/rides"
            >
                        Breckenridge
            </NavLink>
            <div
                        className="pulse"
            />
</div>,
        ],
        "className": "destinations",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="location-pin"
              name="Red Rocks"
              onClick={[Function]}
              to="/rides"
>
              Red Rocks
              <div
                            className="pulse"
              />
</NavLink>,
            "className": "location-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": Array [
                "Red Rocks",
                <div
                  className="pulse"
/>,
              ],
              "className": "location-pin",
              "name": "Red Rocks",
              "onClick": [Function],
              "to": "/rides",
            },
            "ref": null,
            "rendered": Array [
              "Red Rocks",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "pulse",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="location-pin"
                name="Breckenridge"
                onClick={[Function]}
                to="/rides"
>
                Breckenridge
</NavLink>,
              <div
                className="pulse"
/>,
            ],
            "className": "location-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Breckenridge",
                "className": "location-pin",
                "name": "Breckenridge",
                "onClick": [Function],
                "to": "/rides",
              },
              "ref": null,
              "rendered": "Breckenridge",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "pulse",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
